core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -473,6 +491,15 @@ void injectDependencies(BaseCommand baseCommand) {
                             field.setAccessible(true);
                         }
                         field.set(baseCommand, object);
+                        {
```

Comment:
we got rid of the other method handle stuff as it didnt help, but for setters i guess im ok with it if it works without errors.

however, this approach simply does not work.

Above all this code does comparison for ID/Key.

You need to store the key with the type + handle, and the reinjection needs to only apply on matching keys.

Reply:
I think id rather us just use the field, so its easier to look up the fields key in the reinjection.